network/p2p/http.go

```diff
@@ -46,6 +47,15 @@ func MakeHTTPServer(streamHost host.Host) *HTTPServer {
 		Host:       libp2phttp.Host{StreamHost: streamHost},
 		p2phttpMux: mux.NewRouter(),
 	}
+	// libp2phttp server requires either explicit ListenAddrs or streamHost.Addrs() to be non-empty.
+	// If streamHost.Addrs() is empty, we will listen on all interfaces
+	if len(streamHost.Addrs()) == 0 {
+		logging.Base().Debugf("MakeHTTPServer: no addresses for %s, asking to listen all interfaces", streamHost.ID())
```

Comment:
Is there any case where the node runner explicitly does NOT want to listen on all interfaces?

Reply:
it is possible but all interfaces are private so in 99% cases it does not matter unless there is some fancy setup.